File: merge-intervals/merge-intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        intervals.sort(key=lambda x: x[0])
        ret = []
        
        i = 0
        while i < len(intervals):
            cur = intervals[i]
            j = i+1
            while j < len(intervals) and cur[1] >= intervals[j][0]:
                cur[1] = max(cur[1], intervals[j][1])
                j += 1
                i += 1
            ret.append(cur)
            i += 1
        
        return ret

File: merge-intervals/merge-intervals.java
class Solution {
    public int[][] merge(int[][] intervals) {
        Arrays.sort(intervals, Comparator.comparingInt((x) -> x[0]));
        
        var ret = new ArrayList<int[]>();
        
        for (int i=0; i<intervals.length; i++) {
            int start = intervals[i][0], end = intervals[i][1];
            for(int j=i+1; j<intervals.length; j++) {
                if(intervals[j][0] <= end) {
                    end = Math.max(intervals[j][1], end);
                    i = j;
                }
            }
            ret.add(new int[] {start, end});
        }
        var retArr = new int[ret.size()][2];
        for(int i=0; i<retArr.length; i++) {
            retArr[i] = ret.get(i);
        }
        return retArr;
    }
}